File: easy/go/1207-unique-number-of-occurrences.go
package main

func uniqueOccurrences(arr []int) bool{
	occurrenceMap := make(map[int]int)

	for i := 0; i < len(arr); i++ {
		occurrences, ok := occurrenceMap[arr[i]]
		if ok {
			occurrenceMap[arr[i]] = occurrences + 1
		} else {
			occurrenceMap[arr[i]] = 1
		}
	}

	occurrenceSet := make(map[int]bool)

	for _, v := range occurrenceMap {
		_, ok := occurrenceSet[v];

		if ok {
			return false
		} else {
			occurrenceSet[v] = true
		}
	}

	return true
}

File: easy/javascript/1207-unique-number-of-occurrences.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var uniqueOccurrences = function (arr) {
  const occurrenceMap = new Map();

  for (let i = 0; i < arr.length; i++) {
    const occurrences = occurrenceMap.get(arr[i]);

    if (occurrences) {
      occurrenceMap.set(arr[i], occurrences + 1);
    } else {
      occurrenceMap.set(arr[i], 1);
    }
  }

  const occurrenceSet = new Set();

  for (const [_, v] of occurrenceMap) {
    if (occurrenceSet.has(v)) {
      return false;
    } else {
      occurrenceSet.add(v);
    }
  }

  return true;
};
